plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/runtime/RunnableWithResultEx.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jkiss.dbeaver.model.runtime;
+
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * Runnable which stores some result
+ */
+public abstract class RunnableWithResultEx<RESULT_TYPE>
+    extends RunnableWithResult<RunnableWithResultEx.ResultOrError<RESULT_TYPE>> {
+
+    public static class ResultOrError<RESULT_TYPE> { 
+        private final RESULT_TYPE result;
+        private final Exception exception;
+        
+        public ResultOrError(RESULT_TYPE result, Exception exception) {
```

Comment:
Please add nullability annotation to the `exception` parameter.

Earlier I explained why we would not want to have the `result` parameter any nullability annotations at all